# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1506502708 18000
#      Wed Sep 27 03:58:28 2017 -0500
# Node ID cc631e9926e1bdf2c98e44ac7acf903f2a318123
# Parent  730227072acb7e31a7f2716521619be67e8a9026
Properly restore the input stream pointer at end of textscan (bug #52116 et al.)

* oct-stream.cc (delimited_stream::refresh_buf): Do not update stream pointer
  buf_in_file from the current input stream position unless the delimited
  stream buffer is completely empty, e.g., at start of processing.

diff --git a/libinterp/corefcn/oct-stream.cc b/libinterp/corefcn/oct-stream.cc
--- a/libinterp/corefcn/oct-stream.cc
+++ b/libinterp/corefcn/oct-stream.cc
@@ -1470,24 +1470,25 @@ namespace octave
 
     octave_quit ();                       // allow ctrl-C
 
     if (old_remaining > 0)
       {
         buf_in_file += (idx - buf);
         memmove (buf, idx, old_remaining);
       }
+    else
+      buf_in_file = i_stream.tellg ();    // record for destructor
 
     progress_marker -= idx - buf;         // where original idx would have been
     idx = buf;
 
     int gcount;   // chars read
     if (! i_stream.eof ())
       {
-        buf_in_file = i_stream.tellg ();   // record for destructor
         i_stream.read (buf + old_remaining, bufsize - old_remaining);
         gcount = i_stream.gcount ();
       }
     else
       gcount = 0;
 
     eob = buf + old_remaining + gcount;
     last = eob;
